frontends/ol-components/src/components/Card/ListCard.tsx

```diff
@@ -0,0 +1,200 @@
+  height: 122px;
+  margin: 24px 24px 24px 0;
+  border-radius: 4px;
+  object-fit: cover;
+  ${theme.breakpoints.down("md")} {
+    width: 111px;
+    height: 104px;
+    margin: 0;
+    border-radius: 0;
+  }
+
+  background-color: ${theme.custom.colors.lightGray1};
+  flex-shrink: 0;
+`
+
+const Info = styled.div`
+  ${{ ...theme.typography.subtitle3 }}
+  margin-bottom: 16px;
+  ${theme.breakpoints.down("md")} {
+    ${{ ...theme.typography.subtitle4 }}
+    margin-bottom: 8px;
+  }
+
+  color: ${theme.custom.colors.silverGrayDark};
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`
+
+const Title = styled.h3`
+  flex-grow: 1;
+  text-overflow: ellipsis;
+  ${{ ...theme.typography.subtitle1 }}
+  height: ${theme.typography.pxToRem(40)};
+  ${theme.breakpoints.down("md")} {
+    ${{ ...theme.typography.body3 }}
+    height: ${theme.typography.pxToRem(32)};
+  }
+
+  overflow: hidden;
+  margin: 0;
+
+  @supports (-webkit-line-clamp: 2) {
+    white-space: initial;
+    display: -webkit-box;
+    -webkit-line-clamp: 2;
+    -webkit-box-orient: vertical;
+  }
+`
+
+const Footer = styled.span`
+  display: block;
+  ${{
+    ...theme.typography.body3,
+    color: theme.custom.colors.silverGrayDark,
+  }}
+
+  span {
+    color: ${theme.custom.colors.black};
+  }
+
+  white-space: nowrap;
+`
+
+const Bottom = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: flex-end;
+  height: ${theme.typography.pxToRem(32)};
+  ${theme.breakpoints.down("md")} {
+    height: ${theme.typography.pxToRem(18)};
+  }
+`
+
+const Actions = styled.div<{ hasImage?: boolean }>`
+  display: flex;
+  gap: 8px;
+  position: absolute;
+  bottom: 24px;
+  right: ${({ hasImage }) => (hasImage ? "284px" : "24px")};
+  ${theme.breakpoints.down("md")} {
+    bottom: 12px;
+    right: ${({ hasImage }) => (hasImage ? "124px" : "12px")};
+  }
+
+  background-color: ${theme.custom.colors.white};
+`
+
+type CardProps = {
+  children: ReactNode[] | ReactNode
+  className?: string
+  href?: string
+}
+type Card = FC<CardProps> & {
+  Content: FC<{ children: ReactNode }>
+  Image: FC<ImgHTMLAttributes<HTMLImageElement>>
+  Info: FC<{ children: ReactNode }>
+  Title: FC<{ children: ReactNode }>
+  Footer: FC<{ children: ReactNode }>
+  Actions: FC<{ children: ReactNode }>
+}
+
+const ListCard: Card = ({ children, className, href }) => {
+  const _Container = href ? LinkContainer : Container
+
+  let content, imageProps, info, title, footer, actions
+
+  Children.forEach(children, (child) => {
+    if (!isValidElement(child)) return
+    if (child.type === Content) content = child.props.children
+    else if (child.type === Image) imageProps = child.props
+    else if (child.type === Info) info = child.props.children
+    else if (child.type === Title) title = child.props.children
+    else if (child.type === Footer) footer = child.props.children
+    else if (child.type === Actions) actions = child.props.children
+  })
```

Comment:
I'd not seen this pattern before...what is its purpose? Is it so that child order doesn't matter? I.e., so that
```
// 1
<ListCard>
  <ListCard.Content>...</ListCard.Content>
  <ListCard.Actions>...</ListCard.Actions>
</ListCard>
// 2
<ListCard>
  <ListCard.Actions>...</ListCard.Actions>
  <ListCard.Content>...</ListCard.Content>
</ListCard>
```
are treated the same?

Reply:
Yes, the pattern allows us to
- Provide slots that can be used without respect to the sequence
- Insert slots into a nested structure (example we might want to provide more granular slots inside the footer for e.g. start date, learning format)
- Give different treatment depending on the component, e.g. here we use image props if an image is provided - the image doesn't have children.

This RFC with [open PR](https://github.com/reactjs/rfcs/pull/223) goes into the detail: https://github.com/nihgwu/rfcs/blob/neo/slots/text/0000-slots.md